waspc/data/Generator/templates/sdk/wasp/auth/utils.ts

```diff
@@ -81,7 +81,18 @@ export type ProviderId = {
 export function createProviderId(providerName: ProviderName, providerUserId: string): ProviderId {
   return {
     providerName,
-    providerUserId: providerUserId.toLowerCase(),
+    providerUserId: normalizeProviderUserId(providerName, providerUserId)
+  }
+}
+
+// PRIVATE API
+function normalizeProviderUserId(providerName: ProviderName, providerUserId: string): string {
+  switch (providerName) {
+    case 'email':
+    case 'username':
+      return providerUserId.toLowerCase()
```

Comment:
I won't block on this, but it feels for me that it would make more sense to lowercase this inside the email/username code, not here. And then this code should not normalize at all, the provider is the one in charge.